app/src/main/java/com/ataulm/artcollector/gallery/ui/GalleryViewModel.kt

```diff
@@ -1,54 +1,68 @@
 package com.ataulm.artcollector.gallery.ui
 
+import android.view.View
 import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.paging.*
 import com.ataulm.artcollector.Event
+import com.ataulm.artcollector.Painting
 import com.ataulm.artcollector.gallery.domain.GetGalleryUseCase
-import kotlinx.coroutines.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
 import javax.inject.Inject
 
 internal class GalleryViewModel @Inject constructor(
         private val getGallery: GetGalleryUseCase
 ) : ViewModel() {
 
-    private val _gallery = MutableLiveData<UiGallery>()
-    val gallery: LiveData<UiGallery> = _gallery
-
     private val _events = MutableLiveData<Event<NavigateCommand>>()
     val events: LiveData<Event<NavigateCommand>>
         get() = _events
 
     private val parentJob = Job()
     private val coroutineScope = CoroutineScope(parentJob)
 
-    init {
-        coroutineScope.launch(Dispatchers.IO) {
-            val gallery = getGallery()
-            val paintingUis = gallery.map { painting ->
-                UiPainting(
-                        id = painting.id,
-                        title = painting.title,
-                        imageUrl = painting.imageUrl,
-                        artistId = painting.artist.id,
-                        artistName = painting.artist.name,
-                        onClickPainting = { onClick(it) },
-                        onClickArtist = { onClickArtist(it) }
-                )
-            }
-            val uiGallery = UiGallery(paintingUis)
-            withContext(Dispatchers.Main) { _gallery.value = uiGallery }
-        }
+    fun pagedGallery(): Flow<PagingData<UiPainting>> {
+        val pagingSource = getGallery()
+        val pager = Pager(config = PagingConfig(pageSize = 9)) { pagingSource }
```

Comment:
❓ I don't understand why `pagingSource` is returned in a function, instead of being passed in as a constructor parameter.